### PR TITLE
Relax cleanup check in SnapshotStressTestsIT

### PR DESCRIPTION
**Pull request by DaveCTurner**
We can't assert no leaked blobs here because today the first cleanup
leaves the original `RepositoryData` in place so the second cleanup is
not a no-op.

Relates #100718

### CODE REVIEW

File: server/src/internalClusterTest/java/org/elasticsearch/snapshots/SnapshotStressTestsIT.java [hunk 1]
```diff
@@ -801,15 +801,21 @@ private void startCleaner() {
...
+                                // the same repository however should always find no more dangling blobs and be a no-op since we block all
+                                // concurrent operations on the repository.
                                 client.admin()
                                     .cluster()
                                     .prepareCleanupRepository(trackedRepository.repositoryName)
                                     .execute(mustSucceed(secondCleanupRepositoryResponse -> {
                                         final RepositoryCleanupResult secondCleanupResult = secondCleanupRepositoryResponse.result();
-                                        assertThat(Strings.toString(secondCleanupResult), secondCleanupResult.blobs(), equalTo(0L));
-                                        assertThat(Strings.toString(secondCleanupResult), secondCleanupResult.bytes(), equalTo(0L));
+                                        if (secondCleanupResult.blobs() == 1) {
```
Comment (ywangd) [inline server/src/internalClusterTest/java/org/elasticsearch/snapshots/SnapshotStressTestsIT.java:811]: Should we assert that the blob name starts with `index-`?

**Review by ywangd — APPROVED** [review]
LGTM

Have we ever seen a test failure caused by this issue? I have _not_ be able to make it happen locally yet.

**Comment by DaveCTurner** [comment]
Yes, it's very rare to see `SnapshotStressTestsIT` fail at all but if it does then this is one of the problems.

File: server/src/internalClusterTest/java/org/elasticsearch/snapshots/SnapshotStressTestsIT.java (hunk 1 above)
Comment (DaveCTurner) [reply server/src/internalClusterTest/java/org/elasticsearch/snapshots/SnapshotStressTestsIT.java:811]: There isn't really an easy way to do that AFAICT, at least not without adding a bunch more infrastructure. And this isn't a very important assertion in the context of these tests.

**Pull request merged**